R/Vietnam.R

```diff
@@ -0,0 +1,124 @@
+                           "cases_total", "deaths_total", "recovered_total"
+                         ),
+                         #' @field source_text Plain text description of the source of the data
+                         source_text = "Public COVID-19 data curated by 5F team",
+                         #' @field source_url Website address for explanation/introduction of the
+                         #' data
+                         source_url = "https://covid.ncsc.gov.vn", # nolint
+                         
+                         #' @description Set up a table of region codes for clean data
+                         #' @importFrom tibble tibble
+                         set_region_codes = function(){
+                           self$codes_lookup$`1` <- covidregionaldata::vietnam_codes
+                         },
+                         
+                         #' @description Provincial Level Data
+                         #' cleaning
+                         #' @param ... pass additional arguments
+                         #'
+                         #' @importFrom dplyr filter select mutate rename
+                         #' @importFrom tidyr replace_na drop_na
+                         #' @importFrom lubridate dmy 
+                         #' @importFrom jsonlite fromJSON
+                         clean_common = function() {
+                           Sys.setenv("VROOM_CONNECTION_SIZE" = 131072*4) # Fix VROOM error 
+                           provines_url = 'https://covid.ncsc.gov.vn/api/v3/covid/provinces'
+                           bundles = names(self$data$raw)
+                           provines_data = jsonlite::fromJSON(provines_url)
+                           
+                           get_bundles_data = function(bundles){
+                             bundles_data = list()
+                             for (bundle in bundles){ 
+                               url = paste0('https://covid.ncsc.gov.vn/api/v3/covid/provinces?filter_type=', bundle)
+                               data = jsonlite::fromJSON(url)
+                               bundles_data = c(bundles_data, setNames(list(data), bundle)) 
+                             }
+                             bundles_data
+                           }
+                           
+                           bundles_data = get_bundles_data(bundles)
+                           
+                           get_province = function(id, data){
+                             row_dat = provines_data[(id=id),]
+                             death_by_time= do.call(cbind, data$death_by_time[id])
+                             case_by_time=do.call(cbind, data$case_by_time[id]) 
+                             recovered_by_time=do.call(cbind, data$recovered_by_time[id]) 
+                             if (!identical(row.names(death_by_time), row.names(death_by_time))) {
+                               stop("Dates on case_by_time and death_by_time do not match!")
```

Comment:
Does this need to be a fatal error? For some users who may only be using one set of data, they may not care that they don't line up but might be frustrated if they don't get the data they want because some other data they don't care about is having a bad day.

Reply:
I do not know in other cases, but in this one, I am doing `cbind` and dates for `deaths_total` and `cases_total` should match. Otherwise things would be messed up badly.

Reply:
Can you not do an a full join so that dates with only partial data are still included?